Utils/azure-toolkit-ide-libs/azure-toolkit-ide-common-lib/src/main/java/com/microsoft/azure/toolkit/ide/common/component/AzureResourceLabelView.java

```diff
@@ -94,32 +102,19 @@ public void dispose() {
         this.refresher = null;
     }
 
-    public static String initIconPath(IAzureBaseResource<?, ?> resource) {
-        final String formalStatus = resource.getFormalStatus();
-        if (resource instanceof AzResource) {
-            return getIconPath((AzResource<?, ?, ?>) resource);
-        }
-        final String type = resource.getClass().getSimpleName().toLowerCase();
-        final String icon = StringUtils.isBlank(formalStatus) ? type : String.format("%s-%s", type, formalStatus.toLowerCase().trim());
-        return String.format("/icons/%s.svg", icon);
+    @Override
+    public AzureIcon getIcon() {
+        final String iconPath = iconPathFunction.apply(this.resource);
+        final List<AzureIcon.Modifier> modifiers = modifierFunctionList.stream()
+                .map(function -> function.apply(this.resource)).filter(Objects::nonNull).collect(Collectors.toList());
+        return AzureIcon.builder().iconPath(iconPath).modifierList(modifiers).build();
     }
 
-    @Nonnull
-    private static String getIconPath(AzResource<?, ?, ?> resource) {
-        final String status = resource.getStatus();
-        AzResource<?, ?, ?> current = resource;
-        final StringBuilder modulePath = new StringBuilder();
-        while (!(current instanceof AzResource.None)) {
-            modulePath.insert(0, "/" + current.getModule().getName());
-            current = current.getParent();
-        }
-        String fallback = String.format("/icons%s/default.svg", modulePath);
-        if (status.toLowerCase().endsWith("ing")) {
-            fallback = "/icons/spinner";
-        } else if (status.toLowerCase().endsWith("ed")) {
-            fallback = "/icons/error";
-        }
-        final String iconPath = String.format("/icons%s/%s.svg", modulePath, status.toLowerCase());
-        return iconPath + ":" + fallback;
+    public static <T extends IAzureBaseResource<?, ?>> String getResourceIconPath(T resource) {
+        return String.format("/icons/%s.svg", resource.getClass().getSimpleName().toLowerCase());
+    }
+
+    public static <T extends IAzureBaseResource<?, ?>> AzureIcon.Modifier getStatusModifier(T resource) {
+        return new AzureIcon.Modifier(resource.status(), AzureIcon.ModifierLocation.BOTTOM_RIGHT);
```

Comment:
how to support fallback icon?

Reply:
ResourceLabel will just return the icon and the modifiers, the fallback will be implemented in each toolkit, for IntelliJ, it will just use the basic icon without modifer as fallback, [refers](https://github.com/microsoft/azure-tools-for-java/pull/6318/files#diff-1dc92bc441e5c2fbecdfe143896cd7c07bbbb2bd7350ac0e656aa67d1571ce2aR105)